python/ray/experimental/tf_utils.py

```diff
@@ -179,19 +178,42 @@ def set_weights(self, new_weights):
             for name, var in self.variables.items():
                 var.assign(new_weights[name])
         else:
-            assign_list = [
-                self.assignment_nodes[name] for name in new_weights.keys()
-                if name in self.assignment_nodes
-            ]
-            assert assign_list, \
-                "No variables in the input matched those in the network. " \
-                "Possible cause: Two networks were defined in the same " \
-                "TensorFlow graph. To fix this, place each network " \
-                "definition in its own tf.Graph."
-            self.sess.run(
-                assign_list,
-                feed_dict={
-                    self.placeholders[name]: value
-                    for (name, value) in new_weights.items()
-                    if name in self.placeholders
-                })
+            assign_list, feed_dict = self._assign_weights(new_weights)
+            self.sess.run(assign_list, feed_dict=feed_dict)
+
+    def _assign_weights(self, weights):
+        """Sets weigths using exact or closest assignable variable name
+
+        Args:
+            weights (Dict): Dictionary mapping variable names to their
+                weights.
+        Returns:
+            Tuple[List, Dict]: assigned variables list, dict of
+                placeholders and weights
+        """
+
+        assigned = []
+        assignable = set(self.assignment_nodes.keys())
+        feed_dict = {}
+        for name, value in weights.items():
+            close_names = difflib.get_close_matches(name,
```

Comment:
@sven1977 should it be more deterministic? Like by removing path segments of the variable that were removed after 0.8.6. It would make the change less generic though.

Reply:
Can we avoid the dependency on `difflib`?

Reply:
ok, I've removed `difflib` and used a more opinionated comparison based on path segments. I also added iteration on closest matches until we find one that has same shape.